src/main/scala/zio/sqs/SqsStream.scala

```diff
@@ -1,65 +1,39 @@
 package zio.sqs
 
-import software.amazon.awssdk.services.sqs.SqsAsyncClient
-import software.amazon.awssdk.services.sqs.model._
-import zio.stream.Stream
-import zio.{ IO, Task }
-
-import scala.jdk.CollectionConverters._
+import io.github.vigoo.zioaws.sqs
+import io.github.vigoo.zioaws.sqs._
+import io.github.vigoo.zioaws.sqs.model._
+import zio.ZIO
+import zio.stream.ZStream
 
 object SqsStream {
 
   def apply(
-    client: SqsAsyncClient,
     queueUrl: String,
     settings: SqsStreamSettings = SqsStreamSettings()
-  ): Stream[Throwable, Message] = {
-
-    val builder = ReceiveMessageRequest.builder
-      .queueUrl(queueUrl)
-      .attributeNamesWithStrings(settings.attributeNames.asJava)
-      .messageAttributeNames(settings.messageAttributeNames.asJava)
-      .maxNumberOfMessages(settings.maxNumberOfMessages)
-
-    settings.visibilityTimeout.foreach(builder.visibilityTimeout(_))
-    settings.waitTimeSeconds.foreach(builder.waitTimeSeconds(_))
-
-    val request = builder.build
-
-    Stream.fromEffect {
-      Task.effectAsync[List[Message]] { cb =>
-        client
+  ): ZStream[Sqs, Throwable, Message.ReadOnly] = {
+
+    val request = ReceiveMessageRequest(
+      queueUrl,
+      Some(settings.attributeNames),
+      Some(settings.messageAttributeNames),
+      Some(settings.maxNumberOfMessages),
+      settings.visibilityTimeout.map(_.toInt),
```

Comment:
hm.. is `map(_.toInt),` required?
it looks the type is already integer.

--
and actually looking at the settings now, is it worth to replace with integer durations with `zio.duration` type?
but might be worth to make in a separate PR, might be out of scope for this one.

Reply:
One is a scala int, the other is a java one. 

Agreed about zio.duration, but I wanted to keep changes minimal in this PR.

Reply:
Could the java/scala Integer/int thing be addressed in `zio-aws` @vigoo?

Reply:
The goal is that on the zio-aws interfaces everything is Scala friendly, including all primitive and collection types. If in this case you have to pass a boxed java `Integer` that's something that should be fixed in `zio-aws`. I'll take a look tomorrow.

Reply:
Integer fix is going to be published soon: https://github.com/vigoo/zio-aws/pull/120